Content.Server/Doors/Systems/DoorSystem.cs

```diff
@@ -280,6 +282,56 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
         }
     }
 
+    private void OnInserted(EntityUid uid, DoorComponent component, ContainerModifiedMessage args)
+    {
+        if (args.Container.ID != "board" ||
+            !TryComp<AccessReaderComponent>(args.Entity, out var insertedReaderComponent))
+        {
+            return;
+        }
+
+        // We check for 1 and not 0 because the board's already in the container at this point.
+        if (_containerSystem.GetContainer(uid, "board").ContainedEntities.Count != 1)
+        {
+            Logger.Warning($"Attempted to insert a second board {ToPrettyString(args.Entity)} into door {ToPrettyString(uid)}!");
+            return;
+        }
+
+        // If the door didn't have an AccessReader, it does now.
+        var readerComponent = EnsureComp<AccessReaderComponent>(uid);
+
+        if (readerComponent.AccessLists.Count() != 0)
+        {
+            // This door already has access setup on it, so we must be in the
+            // stages of setting up a map. Copy the door's access onto the
+            // board instead.
+            insertedReaderComponent.AccessLists = readerComponent.AccessLists;
+        }
+        else
+        {
+            // Otherwise, copy the board's access onto the door.
+            // This path will be followed if the door had no access to begin
+            // with on map setup, which isn't a problem,
+            // since they'll both be blank.
+            readerComponent.AccessLists = insertedReaderComponent.AccessLists;
+        }
```

Comment:
Having maploader specific codepaths shouldn't be done as it leads to annoying bugs and we've removed most of them. You should just be doing it on accessreader startup, at best just checking lifestage.

Also these need to be deep copied.

Reply:
At present, I'm unable to figure out how to both support inserting a door electronics board that has new access into an airlock and the copying of map access onto a board at map load without somehow making the code aware of map loading, given the door electronics are going through component startup before the doors, and the insertion event is firing on map load. It's a chicken and egg problem.